modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -162,6 +171,9 @@ public class PartitionReplicaLifecycleManager implements IgniteComponent {
 
     private final Set<ReplicationGroupId> replicationGroupIds = ConcurrentHashMap.newKeySet();
 
+    /** (zoneId -> lock) map to provide concurrent access to the zone replicas list. */
+    private final Map<Integer, StampedLock> zonePartitionsLocks = new ConcurrentHashMap<>();
```

Comment:
Where and when this map will be cleared?

Reply:
Never at the moment, my fault. Let's do it under the next https://issues.apache.org/jira/browse/IGNITE-22950 issue, which is already in progress, if you don't mind